crates/ironrdp-cliprdr-format/src/html.rs

```diff
@@ -5,33 +5,42 @@ pub enum HtmlError {
     #[error("invalid CF_HTML format")]
     InvalidFormat,
     #[error("invalid UTF-8")]
-    InvalidUtf8(#[from] std::string::FromUtf8Error),
+    InvalidUtf8(#[from] core::str::Utf8Error),
     #[error("failed to parse integer")]
-    InvalidInteger(#[from] std::num::ParseIntError),
+    InvalidInteger(#[from] core::num::ParseIntError),
     #[error("invalid integer conversion")]
     InvalidConversion,
 }
 
 /// Converts `CF_HTML` format to plain HTML text.
-pub fn cf_html_to_plain_html(input: &[u8]) -> Result<String, HtmlError> {
+///
+/// Note that the `CF_HTML` format is using UTF-8, and the input is expected to be valid UTF-8.
+/// However, there is no easy way to know the size of the `CF_HTML` payload:
+/// 1) it’s typically not null-terminated, and
+/// 2) reading the headers is already half of the work.
+/// Because of that, this function takes the input as a byte slice and finds the end of the payload itself.
+/// This is expected to be more convenient at the callsite.
+pub fn cf_html_to_plain_html(input: &[u8]) -> Result<&str, HtmlError> {
```

Comment:
We were needlessly allocating the output which is a view on the input. It’s better to push the allocation "up".